rimage/image_file.go

```diff
@@ -160,47 +209,26 @@ func SaveImage(pic image.Image, loc string) error {
 
 // DecodeImage takes an image buffer and decodes it, using the mimeType
 // and the dimensions, to return the image.
-func DecodeImage(ctx context.Context, imgBytes []byte, mimeType string, width, height int) (image.Image, error) {
-	return decodeImage(ctx, imgBytes, mimeType, width, height, true)
-}
-
-// decodeImage decodes an image based on its data and MIME type. If checkLazy is true
-// and the MIME type has a lazy suffix, we will not fully decode the image and instead
-// return an intermediate (lazy), not yet decoded image type.
-func decodeImage(
-	ctx context.Context,
-	imgBytes []byte,
-	mimeType string,
-	width, height int,
-	checkLazy bool,
-) (image.Image, error) {
+func DecodeImage(ctx context.Context, imgBytes []byte, mimeType string) (image.Image, error) {
 	_, span := trace.StartSpan(ctx, "rimage::DecodeImage::"+mimeType)
 	defer span.End()
-
-	if checkLazy {
-		actualType, isLazy := ut.CheckLazyMIMEType(mimeType)
-		if isLazy {
-			return NewLazyEncodedImage(imgBytes, actualType, width, height), nil
-		}
+	mimeType, returnLazy := ut.CheckLazyMIMEType(mimeType)
+	if returnLazy {
+		return NewLazyEncodedImage(imgBytes, mimeType), nil
 	}
 
-	switch mimeType {
-	case ut.MimeTypeRawRGBA:
-		img := image.NewNRGBA64(image.Rect(0, 0, width, height))
-		img.Pix = imgBytes
-		return img, nil
-	case ut.MimeTypeJPEG:
-		img, err := jpeg.Decode(bytes.NewReader(imgBytes))
-		return img, err
-	case ut.MimeTypePNG:
-		img, err := png.Decode(bytes.NewReader(imgBytes))
-		return img, err
-	case ut.MimeTypeQOI:
-		img, err := qoi.Decode(bytes.NewReader(imgBytes))
-		return img, err
-	default:
-		return nil, errors.Errorf("do not how to decode MimeType %s", mimeType)
+	img, usedMimeType, err := image.Decode(bytes.NewReader(imgBytes))
+	if err != nil {
+		return nil, err
 	}
+	if ("image/" + usedMimeType) != mimeType {
+		return nil, errors.Errorf(
```

Comment:
This is a behavior change from today. Is this going to prevent camera servers from returning a different mime type from the one that was requested?

Reply:
uh oh, I didn't intend a behavior change. Um, I think I can just remove?

Reply:
yes, i think so